Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -624,12 +630,45 @@ void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* m, CGPoint point, bool eoFill) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (m) {
+        point = CGPointApplyAffineTransform(point, *m);
+    }
+    // check if the point is outside this box already, if it is, return false
+    CGRect boundingBox = CGPathGetBoundingBox(path);
+
+    if ((boundingBox.origin.x > point.x) || (boundingBox.origin.y > point.y) ||
+        ((boundingBox.origin.x + boundingBox.size.width) < point.x) || ((boundingBox.origin.y + boundingBox.size.height) < point.y)) {
+        return false;
+    }
+
+    CGContextRef context =
+        CGBitmapContextCreate(0, boundingBox.origin.x + boundingBox.size.width, boundingBox.origin.y + boundingBox.size.height, 1, 1, 0, 0);
+
+    CGContextSetGrayFillColor(context, 1.0, 1.0);
+    CGContextAddPath(context, path);
+
+    if (eoFill) {
+        CGContextEOFillPath(context);
+    } else {
+        CGContextFillPath(context);
+    }
+
+    CGImageRef image = context->Backing()->DestImage();
+
+    float r = 0.0f;
+    float g = 0.0f;
+    float b = 0.0f;
+    float a = 0.0f;
+    image->Backing()->GetPixel(point.x, point.y, r, g, b, a);
```

Comment:
@bviglietta - No - transforms are handled earlier in the method before any bounds checking at all occurs. This allows for a quick return without having to check further if the point is obviously out of bounds for the path's bounding rect. 
What I found this morning when I looked again was that, for the original implementation that I sent the PR for (where I was checking pixel color), I was setting a context fill color. If we are able to use the cairo functions that shouldn't be needed any more. I removed that, and I think the cairo functions are now working as they should. Still testing some edge cases and reviewing my new implementation though.